### ISSUE TITLE
Versions 4.0.0-4.1.4 Android 6.0 QR code scanning crashes app immediately.

### ISSUE THREAD

**Issue by barkoder** [meshenger-app/meshenger-android#100]
Android 6.0
Versions 4.0.0-4.1.4

Using Android 6.0 to scan a QR code on Android 11.0 crashes the app on Android 6.0.

This happens in all versions from 4.0.0 onwards.

If I downgrade to 3.0.3 on my Android 6.0 and scan the QR Code on Android 11, it doesn't crash. The Android 11.0 contact gets added.  
But if I call the Android 11(running 4.1.4) from the Android 6.0(running 3.0.3), it crashes the app on the Android 11.0.

So basically, this app is not functional.

Please help.

Thanks.

**Comment by mwarning** [comment]
Hi @barkoder 

Meshenger 4.1.0 and up are not compatible with lower versions.
I tried to scan with Android 6.0 and was able to reproduce a crash:

```
FATAL EXCEPTION: DecoderThread
                                                                        Process: d.d.meshenger, PID: 4295
                                                                        java.lang.NoSuchMethodError: No interface method sort(Ljava/util/Comparator;)V in class Ljava/util/List; or its super classes (declaration of 'java.util.List' appears in /system/framework/core-libart.jar)
                                                                        	at com.google.zxing.qrcode.detector.FinderPatternFinder.selectBestPatterns(FinderPatternFinder.java:624)
                                                                        	at com.google.zxing.qrcode.detector.FinderPatternFinder.find(FinderPatternFinder.java:164)
                                                                        	at com.google.zxing.qrcode.detector.Detector.detect(Detector.java:81)
                                                                        	at com.google.zxing.qrcode.QRCodeReader.decode(QRCodeReader.java:77)
                                                                        	at com.google.zxing.MultiFormatReader.decodeInternal(MultiFormatReader.java:173)
                                                                        	at com.google.zxing.MultiFormatReader.decodeWithState(MultiFormatReader.java:87)
                                                                        	at com.journeyapps.barcodescanner.Decoder.decode(Decoder.java:75)
                                                                        	at com.journeyapps.barcodescanner.Decoder.decode(Decoder.java:49)
                                                                        	at com.journeyapps.barcodescanner.DecoderThread.decode(DecoderThread.java:146)
                                                                        	at com.journeyapps.barcodescanner.DecoderThread.access$000(DecoderThread.java:22)
                                                                        	at com.journeyapps.barcodescanner.DecoderThread$1.handleMessage(DecoderThread.java:38)
                                                                        	at android.os.Handler.dispatchMessage(Handler.java:98)
                                                                        	at android.os.Looper.loop(Looper.java:148)
                                                                        	at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by mwarning** [comment]
Downgrading com.google.zxing to 3.3.3 fixes the issue.

**Comment by mwarning** [comment]
@barkoder the new release is out. Please try it out and open the ticket if there are still issues.

**Comment by barkoder** [comment]
Thank you. QR Code was scanned successfully and contact was added but there are still other issues. I'll open a new ticket for those.

Also please never drop support for older Android versions as those devices are the ones I predominantly use for off-the-grid communications in my house.

Thanks again.

**Comment by mwarning** [comment]
Well, I do not have a dedicated Android 6 phone and testing takes time.
Anyway, Android 6.0 support was not dropped, it was a bug. 

Let's see what the other issues are about. :-)

**Comment by mwarning** [comment]
Ah, you mean not to drop Android 6 support in the future. Nah, that's not planned as long I can help it.
